src/plugins/index_pattern_management/public/components/field_editor/lib/validate_script.ts

```diff
@@ -17,72 +17,50 @@
  * under the License.
  */
 
-import { Query } from 'src/plugins/data/public';
 import { HttpStart } from 'src/core/public';
 import { ExecuteScriptParams, ExecuteScriptResult } from '../types';
 
 export const executeScript = async ({
   name,
-  lang,
   script,
   indexPatternTitle,
   query,
   additionalFields = [],
   http,
 }: ExecuteScriptParams): Promise<ExecuteScriptResult> => {
-  // Using _msearch because _search with index name in path dorks everything up
-  const header = {
-    index: indexPatternTitle,
-    ignore_unavailable: true,
-  };
-
-  const search = {
-    query: {
-      match_all: {},
-    } as Query['query'],
-    script_fields: {
-      [name]: {
-        script: {
-          lang,
-          source: script,
-        },
-      },
-    },
-    _source: undefined as string[] | undefined,
-    size: 10,
-    timeout: '30s',
-  };
-
-  if (additionalFields.length > 0) {
-    search._source = additionalFields;
-  }
-
-  if (query) {
-    search.query = query;
-  }
-
-  const body = `${JSON.stringify(header)}\n${JSON.stringify(search)}\n`;
-  const esResp = await http.fetch('/elasticsearch/_msearch', { method: 'POST', body });
-  // unwrap _msearch response
-  return esResp.responses[0];
+  return http
+    .post('/internal/index-pattern-management/preview_scripted_field', {
+      body: JSON.stringify({
+        index: indexPatternTitle,
+        name,
+        script,
+        query,
+        additionalFields,
+      }),
+    })
+    .then((res) => ({
+      status: res.statusCode,
+      hits: res.body.hits,
+    }))
+    .catch((err) => ({
+      status: err.statusCode,
+      error: err.body.attributes.error,
+    }));
```

Comment:
Restructured the response to minimize the amount of changes I needed to make to the existing types that the UI relies on.